### PR TITLE
Logfault header only library

### PR DESCRIPTION
**Pull request by jgaa**
This is my first submission to the conan-center-index.

Specify library name and version:  **logfault/0.5.0**

I am the author of the package Logfault. It has some users, so in order
to make it simpler to use for everyone, I'm submitting this PR to include it
in the ConanCenter.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with conan 2.1.0

### CODE REVIEW

File: recipes/logfault/all/conanfile.py
Comment (RubenRBS) [inline recipes/logfault/all/conanfile.py:30]:
```suggestion
```
Not needed, `CMakeToolchain` is already being instantiated in the `generate` method, and `CMakeDeps` is only useful when there are dependencies in the recipe :)

File: recipes/logfault/all/conanfile.py
Comment (RubenRBS) [inline recipes/logfault/all/conanfile.py:43]:
```suggestion
            "msvc": "193",
```
As per https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B, VS 17 is msvc 193

File: recipes/logfault/all/conanfile.py
Comment (RubenRBS) [inline recipes/logfault/all/conanfile.py:54]:
```suggestion
    # Use the export_sources(self) method instead of the exports_sources attribute.
    # This allows finer grain exportation of patches per version
    def export_sources(self):
        export_conandata_patches(self)
```

Not needed if there are no patches

File: recipes/logfault/all/conanfile.py
Comment (RubenRBS) [inline recipes/logfault/all/conanfile.py:83]:
```suggestion
```

Leftover from the template, seems like your library does suport Windows just fine

File: recipes/logfault/all/conanfile.py
Comment (RubenRBS) [inline recipes/logfault/all/conanfile.py:92]:
```suggestion
    def build(self):
```

**Review by RubenRBS — COMMENTED** [review]
Hi! Thanks a lot for yout contribution and taking the time to submit your library to CCI, we appreciate it :)

The recipe looks great overall, the only thing on my side besides the minor review comments, would be to remoce the comments from the template to keep the recipe more readable, thanks!

File: recipes/logfault/all/conanfile.py
Comment (jgaa) [reply recipes/logfault/all/conanfile.py:92]: Thank you for the review!

I will go over it in detail and push a new revision later today (I'm in Europe).

File: recipes/logfault/all/conanfile.py
Comment (uilianries) [inline recipes/logfault/all/conanfile.py:24]:
```suggestion
```

Duplicated with CMakeToolchain class usage.

File: recipes/logfault/all/conanfile.py
Comment (uilianries) [inline recipes/logfault/all/conanfile.py:4]:
```suggestion
from conan.tools.files import copy, get
```

Not used imports

File: recipes/logfault/all/conanfile.py
Comment (uilianries) [inline recipes/logfault/all/conanfile.py:91]:
```suggestion
```
The cmake_find_package and cmake_find_package_multi are not tested/required for new recipes in Conan Center Index: https://github.com/conan-io/conan-center-index/discussions/19104

File: recipes/logfault/all/conanfile.py
Comment (uilianries) [inline recipes/logfault/all/conanfile.py:37]:
```suggestion
           "Visual Studio": "15",
           "msvc": "191",
           "gcc": "8",
           "clang": "7",
           "apple-clang": "12",
```
C++ 17 is supported by older compilers, you can lower the bar for sure.

**Review by uilianries — COMMENTED** [review]
Thank you for your PR! Please, take a look on my review.

File: recipes/logfault/all/conanfile.py
Comment (jgaa) [reply recipes/logfault/all/conanfile.py:24]: Are you sure? I removed it yesterday, and that broke the build. Adding it back fixed the build.

Comment (uilianries) [reply recipes/logfault/all/conanfile.py:24]: Yes, I'm sure. At line [41](https://ci-conan-stg.jfrog.team/job/staging-v1/job/cci/view/change-requests/job/PR-1197/1/console) has CMakeToolchain declared already.

File: recipes/logfault/all/conanfile.py
Comment (uilianries) [inline recipes/logfault/all/conanfile.py:42]:
```suggestion
        tc.variables['LOGFAULT_BUILD_TESTS'] = not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
        tc.generate()
```
That's why was not working. Without generate, the CMakeToolchain will not create the file.

**Comment by jgaa** [comment]
The testing issue seems to be a problem with the logfault source code. 
I am testing with some changes. When that is ready, I will push a new revision here, pointing to the updated tarfile.

**Comment by RubenRBS** [comment]
Thanks, feel free to ping me once ready :)

**Comment by jgaa** [comment]
Is it possible to change this PR to target 0.5.1 in stead of 0.5.0?

If not, I can change the version back in `conandata.yml` and keep the url to the newer tarball so we can complete this review.

**Comment by uilianries** [comment]
@jgaa Hello! As mentioned by the hook error message above:

> [HOOK - conan-center.py] pre_export(): ERROR: [CONFIG.YML HAS NEW VERSION (KB-H052)] The version "0.5.1" exists in "conandata.yml" but not in "../config.yml", so it will not be built. Please update "../config.yml" to include newly added version "0.5.1". (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H052-CONFIG.YML-HAS-NEW-VERSION) 

Both `all/conandata.yml` and `config.yml` should have the very same version listed. Right now, they are different. Could you please update the `config.yml` to `0.5.1`, so they will be aligned.

**Comment by jgaa** [comment]
I can't find any examples of other projects using gtest and cmake.build() for the target package in conan-center-index. 

The problem with the build is that somehow the configuring of gtest does not produce (or appear so to me - I don't have the complete build logs) one of the expected CMake targets; GTest::gtest or gtest::gtest. 

When I run the build locally with testing enabled, I get a message in output: `Conan: Target declared 'gtest::gtest'`. In the build from Conan Center, I don't see this. Just `Found GTest: /home/conan/workspace/prod-v2/bsr/4772/dadbb/p/gtestf28c74556dabb/p/lib/libgtest.a`. 

For now I'm keeping the test code in `conanfile.py`, but disabling it so the project can pass. I would be good however to find the root cause so the unit tests can be ran while building at the Conan Center.

File: recipes/logfault/all/conanfile.py
Comment (valgur) [inline recipes/logfault/all/conanfile.py:72]:
```suggestion
        cmake = CMake(self)
        cmake.configure()
        cmake.build()
```
Running tests, even optionally, is discouraged on CCI.

File: recipes/logfault/all/test_package/CMakeLists.txt
Comment (valgur) [inline recipes/logfault/all/test_package/CMakeLists.txt:10]:
```suggestion
add_executable(${PROJECT_NAME} test_package.cpp)
target_link_libraries(${PROJECT_NAME} PRIVATE logfault::logfault)
target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)
```

File: recipes/logfault/config.yml
Comment (valgur) [inline recipes/logfault/config.yml:2]:
```suggestion
```

File: recipes/logfault/all/conanfile.py
Comment (valgur) [inline recipes/logfault/all/conanfile.py:72]:
```suggestion
```
Running tests, even optionally, is discouraged on CCI.

**Review by valgur — APPROVED** [review]
Thanks! LGTM. Only some minor change suggestions.

File: recipes/logfault/all/conanfile.py
Comment (jgaa) [reply recipes/logfault/all/conanfile.py]: I have removed the option to run unit-tests.

**Review by valgur — APPROVED** [review]

**Comment by jgaa** [comment]
Are there any pending tasks from my side?

**Comment by jgaa** [comment]
Ping @RubenRBS 
Are you guys waiting for anything more from me?

**Comment by RubenRBS** [comment]
Hi @jgaa thanks for the ping

This PR is just waiting for a review from us again - we were out on easter holidays this last week so we didn't get to do much reviewing, I'll try to get this done today :)

File: recipes/logfault/all/conanfile.py
Comment (RubenRBS) [inline recipes/logfault/all/conanfile.py:42]: We're not building the library, so this is probably not needed. Did you do this to be able to call cmake on your own?

**Review by RubenRBS — COMMENTED** [review]
Looks good, minor question about the `generate()` method :)

File: recipes/logfault/all/conanfile.py
Comment (jgaa) [reply recipes/logfault/all/conanfile.py:42]: I used it only for testing, and without tests, it's not needed.
I have pushed an update without the generate method.

**Comment by jgaa** [comment]
Ping @RubenRBS 
Are you guys waiting for any changes from me?

**Comment by RubenRBS** [comment]
Hi @jgaa sorry for the radio silence. There's nothing else needed from you right now, the CI systems will pick the PR up once they leave maintenance mode, thanks!

**Review by RubenRBS — APPROVED** [review]

**Review by toge — APPROVED** [review]

**Pull request merged**
